x-pack/plugins/actions/server/builtin_action_types/xmatters.ts

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { curry, isString } from 'lodash';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+import { ActionsConfigurationUtilities } from '../actions_config';
+import { Logger } from '../../../../../src/core/server';
+import { postXmatters } from './lib/post_xmatters';
+
+export type XmattersActionType = ActionType<
+  ActionTypeConfigType,
+  ActionTypeSecretsType,
+  ActionParamsType,
+  unknown
+>;
+export type XmattersActionTypeExecutorOptions = ActionTypeExecutorOptions<
+  ActionTypeConfigType,
+  ActionTypeSecretsType,
+  ActionParamsType
+>;
+
+const configSchemaProps = {
+  urlConfig: schema.maybe(schema.string()),
+  usesBasic: schema.boolean({ defaultValue: true }),
+};
+const ConfigSchema = schema.object(configSchemaProps);
+export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
+
+// secrets definition
+export type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
+const secretSchemaProps = {
+  user: schema.nullable(schema.string()),
+  password: schema.nullable(schema.string()),
+  urlSecrets: schema.maybe(schema.string()),
+};
+const SecretsSchema = schema.object(secretSchemaProps, {
+  validate: (secrets) => {
+    // user and password must be set together (or not at all)
+    if (!secrets.password && !secrets.user) return;
+    if (secrets.password && secrets.user) return;
+    return i18n.translate('xpack.actions.builtin.xmatters.invalidUsernamePassword', {
+      defaultMessage: 'Both user and password must be specified',
+    });
+  },
+});
+
+// params definition
+export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+const ParamsSchema = schema.object({
+  alertActionGroupName: schema.maybe(schema.string()),
+  signalId: schema.maybe(schema.string()),
+  ruleName: schema.maybe(schema.string()),
+  date: schema.maybe(schema.string()),
+  severity: schema.string(),
+  spaceId: schema.maybe(schema.string()),
+  tags: schema.maybe(schema.string()),
+});
+
+export const ActionTypeId = '.xmatters';
+// action type definition
+export function getActionType({
+  logger,
+  configurationUtilities,
+}: {
+  logger: Logger;
+  configurationUtilities: ActionsConfigurationUtilities;
+}): XmattersActionType {
+  return {
+    id: ActionTypeId,
+    minimumLicenseRequired: 'gold',
```

Comment:
Hey! Is this the correct license for xMatters? @pmuellr

Reply:
Good question, not sure.  @alexfrancoeur ??? I'd guess it is - can't be basic, probably don't want to go as far as platinum ...

Reply:
after some discussion, everyone agrees gold is the license we should be using now